query.c

```diff
@@ -1640,6 +1694,13 @@ query_add_optional(query_type *q, nsd_type *nsd)
 				/* nsid payload */
 				buffer_write(q->packet, nsd->nsid, nsd->nsid_len);
 			}
+			/* Append Extended DNS Error (RFC8914) option if needed */
+			if (q->edns.ede >= 0) { /* < 0 means no EDE */
+				buffer_write_u16(q->packet, 15); /* OPTION-CODE */
+				buffer_write_u16(q->packet,  2 + q->edns.ede_text_len - 1); /* OPTION-LENGTH */
+				buffer_write_u16(q->packet, q->edns.ede); /* INFO-CODE */
+				buffer_write(q->packet, q->edns.ede_text, q->edns.ede_text_len - 1); /* EXTRA-TEXT */
```

Comment:
Voor als `ede_text_len == 0`, als er dus geen tekst is:
```suggestion
				if (q->edns.ede_text_len > 1)
					buffer_write(q->packet, q->edns.ede_text, q->edns.ede_text_len - 1); /* EXTRA-TEXT */
```